src/Mvc/test/Mvc.IntegrationTests/TryUpdateModelIntegrationTest.cs

```diff
@@ -1139,6 +1139,84 @@ public async Task TryUpdateModel_ExistingModelWithNoParameterlessConstructor_Ove
             Assert.Equal(ModelValidationState.Valid, state.ValidationState);
         }
 
+        private record AddressRecord(string Street, string City)
+        {
+            public string ZipCode { get; set; }
+        }
+
+        [Fact]
+        public async Task TryUpdateModel_RecordTypeModel_DoesNotOverwriteConstructorParameters()
```

Comment:
Whoa, this violates the principle of least surprise. Devs likely expect the requested updates to be honoured and, because you're finding the validators for the properties in the constructor parameters, it _could_ work just fine.

Reply:
Let's bring this up doing the design review. We could tweak this if we choose to. I'd thought it would be more surprising to change what are considered immutable properties